desktop/cmp/contextmenu/ContextMenu.ts

```diff
@@ -6,14 +6,28 @@
  */
 import {hoistCmp} from '@xh/hoist/core';
 import '@xh/hoist/desktop/register';
-import {menu, menuDivider, menuItem} from '@xh/hoist/kit/blueprint';
+import {menu, menuDivider, MenuDivider, menuItem, MenuItem} from '@xh/hoist/kit/blueprint';
 import {wait} from '@xh/hoist/promise';
 import {filterConsecutiveMenuSeparators} from '@xh/hoist/utils/impl';
 import {isEmpty} from 'lodash';
-import PT from 'prop-types';
-import {isValidElement} from 'react';
+import {isValidElement, ReactNode} from 'react';
 import {ContextMenuItem} from './ContextMenuItem';
 
+export interface ContextMenuProps {
+    menuItems: ContextMenuItemLike[]
+}
+
+/**
+ * The special string token '-' will be replaced with a MenuDivider.
+ * ReactNodes or other strings will be interpreted as the `text` property for a MenuItem.
+ */
+export type ContextMenuItemLike = ContextMenuItem|
```

Comment:
would we want to import blueprints API with the BP prefix, and name our own just `MenuItem`.